lib/hw_unique_key/Kconfig

```diff
@@ -30,12 +31,13 @@ config HW_UNIQUE_KEY_LOAD
 config HW_UNIQUE_KEY
 	bool "Hardware Unique Keys (HUK)"
 	depends on HAS_HW_NRF_CC310 || HAS_HW_NRF_CC312
-	depends on NRF_CC3XX_PLATFORM
+	depends on NRF_CC3XX_PLATFORM || BUILD_WITH_TFM
 	depends on NRF_SECURITY
-	depends on !ARM_NONSECURE_FIRMWARE
-	depends on MPU_ALLOW_FLASH_WRITE
+	depends on MPU_ALLOW_FLASH_WRITE || BUILD_WITH_TFM
+	depends on !BUILD_WITH_TFM || TFM_CRYPTO_BUILTIN_KEYS
```

Comment:
Why do we have here two times `OR BUILD_WITH_TFM`  and one with`NOT BUILD_WITH_TFM`

Shouldn't the `default y if BUILD_WITH_TFM` be enough?  as all three cases here don't make a difference?

Reply:
the `OR BUILD_WITH_TFM` is the get rid of dependencies when TF-M is enabled, the `NOT BUILD_WITH_TFM` is to add dependencies when TF-M is enabled.

An alternative to this is to instead add it all as a separate entry:

```
config HW_UNIQUE_KEY
	bool "Hardware Unique Keys (HUK)"
	depends on HAS_HW_NRF_CC310 || HAS_HW_NRF_CC312
	depends on NRF_CC3XX_PLATFORM
	depends on NRF_SECURITY
	depends on !ARM_NONSECURE_FIRMWARE
	depends on MPU_ALLOW_FLASH_WRITE
        select FPROTECT if HAS_HW_NRF_ACL
	imply NRFX_NVMC if HAS_HW_NRF_ACL
	
if BUILD_WITH_TFM
config HW_UNIQUE_KEY
      bool "Hardware Unique Key HUK"
      depends on TFM_CRYPTO_BUILTIN_KEYS
      default y
```

Reply:
Ahh ok I see, thanks for explaining.